.github/workflows/check-pr-scalars.yml

```diff
@@ -0,0 +1,77 @@
+name: Check PR scalars
+
+on:
+    workflow_dispatch:
+    push:
+        branches: ['main']
+        paths:
+            - 'packages/scalars/**'
+    pull_request:
+        branches: ['main']
+        paths:
+            - 'packages/scalars/**'
+        types: [opened, synchronize]
+
+env:
+  NX_ISOLATE_PLUGINS: false
+
+jobs:
+    lint:
+        name: Lint
+        runs-on: ubuntu-latest
+        container: node:22
```

Comment:
Adding a comment here but it's not for this PR, I've created a separate issue: https://github.com/powerhouse-inc/powerhouse/issues/371

Each job is running pnpm install in it's own docker instance. And I guess now this runs in all the packages of the monorepo right? We need to be careful in the CI actions to do as less work as possible.

Pnpm has a way to cache the dependencies as we have used in some actions: https://pnpm.io/continuous-integration#github-actions
Here we use `container: node.22` so each jobs runs in its own docker container AFAIK. This way the pnpm cache is not shared between jobs and `nx` is not able to optimize the jobs with cache.

I think we should still use docker actions, as we have add some issues with node APIs not being available in some jobs.

I don't know if there is an alternative approach to this, but using a cache action would already take us far: https://github.com/actions/cache

One idea here is to have a `setup` job that installs and caches the dependencies, using a checksum of `pnpm-lock` as cache key and caching `pnpm store path`.

Then all the other jobs run after it in parallel and load from the cache.

Maybe we could also share the nx cache?